src/diffusers/models/transformer_2d.py

```diff
@@ -81,18 +82,22 @@ def __init__(
         num_attention_heads: int = 16,
         attention_head_dim: int = 88,
         in_channels: Optional[int] = None,
+        out_channels: Optional[int] = None,
         num_layers: int = 1,
         dropout: float = 0.0,
         norm_num_groups: int = 32,
         cross_attention_dim: Optional[int] = None,
         attention_bias: bool = False,
         sample_size: Optional[int] = None,
         num_vector_embeds: Optional[int] = None,
+        patch_size: Optional[int] = None,
         activation_fn: str = "geglu",
         num_embeds_ada_norm: Optional[int] = None,
         use_linear_projection: bool = False,
         only_cross_attention: bool = False,
         upcast_attention: bool = False,
+        use_ada_layer_norm_zero: bool = False,
```

Comment:
Instead of making this a boolean, could we make it a string to be more general for the future? 
```suggestion
        norm_type: str = "layer", # ada_norm, ada_norm_zero
```